retis-events/src/display.rs

```diff
@@ -45,14 +50,175 @@ impl DisplayFormat {
     }
 }
 
+/// `Formatter` implements `std::fmt::Write` and controls how events are being
+/// displayed.  This is similar to `std::fmt::Formatter` but with our own
+/// constraints.
+///
+/// It supports the following capabilities: indentation and itemization. Each of
+/// those are always context-based: the capabilities and their configuration can
+/// change over time and might end based on input (eg. itemization).
+pub struct Formatter<'a, 'inner> {
+    inner: &'a mut fmt::Formatter<'inner>,
+    pub conf: FormatterConf,
+    level: usize,
+    first: bool,
+    start: bool,
```

Comment:
`first` and `start` are kind of similar words to me. It would be nice to document what they are for so that future readers of this struct have a better experience.